### ISSUE TITLE
maven-metadata.xml generated for all directories.

### ISSUE THREAD

**Issue by LexManos** [dzikoysk/reposilite#461]
https://repo.panda-lang.org/releases/com/maven-metadata.xml
This should not be the case. Perhaps limit the xml generation to places that the XML already exists?

**Comment by dzikoysk** [comment]
Reposilite doesn't trust locally stored/deployed metadata files at all, so it's kinda problematic. Metadata is always generated during the first request and then cached in metadata service. Thanks to that mechanism we can limit annoying issues related to desync. Did you find any potential issues caused by this?

**Comment by LexManos** [comment]
Other then the fact that it provides bad xml files for directory listings even if we wanted too not allow browsing?
I get that you generate the xml for one reason or another. But providing random dirrectory listings via maven-metadata.xml does not seem like a good idea.
Thats why I suggested that we limit it to the places that the xmls already exist. That way you don't get a XML for 'com' but you would for anything uploaded.

**Comment by dzikoysk** [comment]
I mean it depends on what you mean by maven-metadata file. From the specification point of view it's just a way to list the given directory to process further requests, it's just like an enhanced version of indexing and it's not random tbh. On the other hand if you look at it like some kind of artifact/build description then yes, it may contains misleading data. I think we can provide the possibility in configuration to reuse existing maven-metadata files for ppl who don't want to use this mechanism to avoid this kind of scenarios, but I'm still pretty sure that most of users should keep it enabled by default.

**Comment by LexManos** [comment]
Well [The docs](https://maven.apache.org/ref/3.2.5/maven-repository-metadata/) define what they should contain. And many projects use them as version lists not directory lists.
So i'd argue it's more on spec to only provide them as specified:
In an artifact's directory listing all versions
In snapshot directories listing all versions of the snapshot.
Haven't dealt with maven plugins so don't know what that format looks like.


I mean, I liked the generation of the XML because it means I could just delete a version and it'd auto-fix the xml without me having to manually do it. But i'd live with that to have my directories not be indexable.

**Comment by dzikoysk** [comment]
I'd argue about the quality of the docs, because it's more like a preview of what you can expect rather than actual technical specification. It's great to read and get some information about what the maven-metadata can be, but I don't think someone will be able to implement repository based on the data provided in docs. Maven team didn't really provide full specification, tbh they asked me if I could create one. The issue about the possibility of indexing is not really related to this topic, but I see your point. Let's link it with #435 to be sure that we won't forget to limit generator in such cases.

**Comment by ldegen** [comment]
Hi there,
I am encountering a problem that *may* be related to this issue.
I am writing a shell script that is trying to determine the current version of the POM by running

```
mvn help:evaluate -Dexpression=project.version
```
I am getting the following error from maven:
```
No plugin found for prefix 'help' in the current project and in the plugin groups [org.apache.maven.plugins, org.codehaus.mojo] available from the repositories
```
I am trying hard to figure out what the actual problem might be. I do not entirely understand what is happening, but it seems, that maven is trying to download 
`https://<my reposilte mirror>/org/apache/maven/plugins/maven-metadata.xml` to figure out what plugins are available for the `help` prefix? It then gets confused by what it finds - the content looks somewhat weird; it has actual plugin-names at places where I think should be version strings -- that's what brought me here, actually. So, reposilite generates the content of this file on the fly?  

I am sorry, I do not understanding half of it. Do you think my problem could be related?

**Comment by ldegen** [comment]
Just a quick follow-up: When copy the contents from the original repository over to my local repository (overwriting the cached version in `$HOME/.m2/repositories/org/apache/maven/plugins/maven-metadata-<my reposilite mirror>`.xml, the `mvn help:evaluate` command executes just fine. So there must be some relation, no?

**Comment by dzikoysk** [comment]
I don't know this plugin, so I can't really tell what's going on with it. I think it could be related to this issue if maven-help-plugin performs some non-standard operations to browse the repository structure. I think we should move this thread to a new issue to investigate this topic.

**Comment by dzikoysk** [comment]
Addressed by #550
